scrapy/downloadermiddlewares/robotstxt.py

```diff
@@ -75,24 +97,27 @@ def robot_parser(self, request, spider):
                 meta={"dont_obey_robotstxt": True},
                 callback=NO_CALLBACK,
             )
+            assert self.crawler.engine
+            assert self.crawler.stats
```

Comment:
I wonder how we could make this easier for users in their own code.

Would mypy understand if we moved this to `__init__` or `from_crawler`?
Is there anything we can do to enable writing a single line that asserts that the crawler has been initialized?

Reply:
Yeah, it would be good if we didn't need that. `from_crawler` is unlikely to help as checkers don't know it was called. Not sure about `__init__`, as checkers could think that the attribute became `None` since then.

Reply:
Dping this in `__init__` doesn't help here, I'll create a separate issue for this (but I'm not sure if it's possible).